### ISSUE TITLE
Current npm version "dslink@2.1.10-beta.0" - exisiting projects fail to run

### ISSUE THREAD

**Issue by ghost** [IOT-DSA/sdk-dslink-ts#19]
Hi,

Not sure how recent this change is, but code that worked with dslink@2.1.5 fails with the lastest NPM release "dslink@2.1.10-beta.0". Running the node.js example given in the readme fails with the following error:-

```
/Users/ross/Documents/mockLink/node_modules/dslink/js/src/responder/base-local-node.js:6
class BaseLocalNode extends node_state_1.LocalNode {
                                         ^
TypeError: Class extends value undefined is not a constructor or null
```
Reverting back to release 2.1.5 fixes the issue

Thanks 

Ross

**Comment by rinick** [comment]
I'm not able to reproduce that issue
I just published version 2.1.14, could you check if that works for you? if that still doesn't work, could you send me a testcase for it?

**Comment by ghost** [comment]
Hi,

Updating to 2.1.14 seems to have addressed the issue.
Thanks for sorting this so quickly.

Kind regards

Ross

**Comment by ghost** [comment]
Sorry I spoke too soon - I was updating the npm package over an existing one. I have spotted an issue with a clean install.

The npm 2.1.4 release solves the original issue above, but seems to be missing dependencies (primarily 'ws').

Test scenario below:

1. Create project folder.
2. Install older SDK version using 

> npm install --save dslink@2.1.5

Run the SDK example as below

`const {DSLink, RootNode, ValueNode} = require("dslink");

class MyValueNode extends ValueNode {
  constructor(path, provider) {
    super(path, provider, 'myvalue', 'number');
    this._value = 123;
  }
}

function main() {
  let rootNode = new RootNode();
  rootNode.createChild('value', MyValueNode);

  let link = new DSLink('mydslink', {rootNode});
  link.connect();
}

main();`

Example runs as expected.

3. Repeat test but install lastest DSlink version 2.1.14 using 

> npm install --save dslink

4. Example code will fail to run with following error:-
`internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'ws'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/Users/ross/Documents/testLink/node_modules/dslink/js/src/nodejs/client-link.js:9:30)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)`

5. install 'ws' using 

> npm install --save ws

6. Code will now run as expected

**Comment by rinick** [comment]
ws package is fixed in 2.1.15
